shell/common/gin_converters/frame_converter.cc

```diff
@@ -18,4 +26,56 @@ v8::Local<v8::Value> Converter<content::RenderFrameHost*>::ToV8(
   return electron::api::WebFrameMain::From(isolate, val).ToV8();
 }
 
+// static
+v8::Local<v8::Value>
+Converter<gin_helper::AccessorValue<content::RenderFrameHost*>>::ToV8(
+    v8::Isolate* isolate,
+    gin_helper::AccessorValue<content::RenderFrameHost*> val) {
+  content::RenderFrameHost* rfh = val.Value;
+  if (!rfh)
+    return v8::Null(isolate);
+
+  const int process_id = rfh->GetProcess()->GetID();
+  const int routing_id = rfh->GetRoutingID();
+
+  if (rfh_templ.IsEmpty()) {
+    v8::EscapableHandleScope inner(isolate);
+    v8::Local<v8::ObjectTemplate> local = v8::ObjectTemplate::New(isolate);
+    local->SetInternalFieldCount(2);
+    rfh_templ.Reset(isolate, inner.Escape(local));
+  }
+
+  v8::Local<v8::Object> rfh_obj =
+      v8::Local<v8::ObjectTemplate>::New(isolate, rfh_templ)
+          ->NewInstance(isolate->GetCurrentContext())
+          .ToLocalChecked();
+
+  rfh_obj->SetInternalField(0, v8::Number::New(isolate, process_id));
+  rfh_obj->SetInternalField(1, v8::Number::New(isolate, routing_id));
+
+  return rfh_obj;
+}
+
+// static
+bool Converter<gin_helper::AccessorValue<content::RenderFrameHost*>>::FromV8(
+    v8::Isolate* isolate,
+    v8::Local<v8::Value> val,
+    gin_helper::AccessorValue<content::RenderFrameHost*>* out) {
+  v8::Local<v8::Object> rfh_obj;
+  if (!ConvertFromV8(isolate, val, &rfh_obj))
+    return false;
+
+  const int process_id =
+      v8::Local<v8::Number>::Cast(rfh_obj->GetInternalField(0))->Value();
```

Comment:
two things:

1. prefer `.As`, e.g. `rfh_obj->GetInternalField(0).As<v8::Number>()->Value()`, as it's a bit easier to read, and
2. We should check that the object (a) has at least two internal fields and (b) that both are v8::Number. `.As` / `Cast` does no runtime checking and will simply corrupt memory if the underlying data isn't the expected type.

Reply:
Are you good with using a `DCHECK`? I was seeing that used pretty commonly throughout chromium for checking internal field counts. aaa6c8f

Reply:
I think at least a `CHECK` is probably warranted here as I'm not sure it's guaranteed that it's impossible to call this on an object we didn't expect it to be called on? e.g. it seems plausible that JS could set up a situation such that `rfh_obj` was an arbitrary v8 object.

It might be nicer to just `if (!...) return false` though.

Reply:
Added type checking f6c9671

Reply:
Went with a conditional for checking field counts to be safe 😅 a45bad2